### PR TITLE
Add unit tests for admin menu state and data layer

### PR DESCRIPTION
**Pull request by getdave**
Given how easy they are to write I added a few unit tests to the implementation of state and data layer.

Will need rebase once https://github.com/Automattic/wp-calypso/pull/45576 has landed.

#### Changes proposed in this Pull Request

* Add unit tests to admin-menu state reducers and actions.
* Add unit test to admin-menu data layer

#### Testing instructions

* Checkout PR
* State tests - `yarn run test-client:watch client/state/admin-menu/test`.
* Data layer tests - `yarn run test-client:watch client/state/data-layer/wpcom/sites/admin-menu`

### CODE REVIEW

File: client/state/admin-menu/test/reducer.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import deepFreeze from 'deep-freeze';
```
Comment (cpapazoglou) [inline client/state/admin-menu/test/reducer.js:4]: Can we avoid using deepFreeze here? It seems that it is not needed. I may be mistaken though!

**Review by cpapazoglou — APPROVED** [review]
LGTM, I left a minor comment regarding the usefulness of `deep-freeze`

File: client/state/admin-menu/test/reducer.js (hunk 1 above)
Comment (getdave) [reply client/state/admin-menu/test/reducer.js:4]: It's a pattern in Calypso to deepFreeze the state in a reducer test in order to check that the reducer doesn't _mutate_ the state.

**Comment by getdave** [comment]
Note: the e2e test failures are unrelated to adding these unit tests.

File: client/state/admin-menu/test/reducer.js (hunk 1 above)
Comment (cpapazoglou) [reply client/state/admin-menu/test/reducer.js:4]: Thanks for clarifying it!

**Pull request merged**